lib/paperclip/tempfile_factory.rb

```diff
@@ -15,7 +13,7 @@ def extension
     end
 
     def basename
-      File.basename(@name, extension).gsub(ILLEGAL_FILENAME_CHARACTERS, '_')
+      SecureRandom.hex(16).encode('UTF-8')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.